File: solutions/java/mkuser.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class mkuser {

    public static void main(String[] args) {

		try {
			if(!(args.length == 4)) {
				System.out.println("syntax: mkuser username password role \"real name\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        User user = service.getUsers().create(args[0], args[1], args[2]);
	        user.setRealName(args[3]);
	        user.update();
	        System.out.println("User " + user.getName() + " created.");
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: solutions/python/mkuser.py

import splunklib.client as client
import utils, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 5:
    print "Syntax: mkuser.py username password role \"real name\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
    params = {"realname": sys.argv[4]}

    user = service.users.create(sys.argv[1], sys.argv[2], sys.argv[3], **params )
    print "User %s created." % user.name

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
